### PR TITLE
Add `replace_in_file` command to change occurrences of text in a file

### PR DESCRIPTION
**Pull request by bfalans**
### Background
See #3643 

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes. 
- [x] I have run the following commands against my code to ensure it passes our linters:
    ```shell
    black .
    isort .
    mypy
    autoflake --remove-all-unused-imports --recursive --ignore-init-module-imports autogpt tests --in-place
    ```

### CODE REVIEW

File: autogpt/commands/file_operations.py [hunk 1]
```diff
@@ -224,6 +224,68 @@ def write_to_file(filename: str, text: str, config: Config) -> str:
...
 
 
+@command(
+    "update_file",
+    "Update file",
+    '"filename": "<filename>",'
+    '"old_text": "<old_text>","new_text": "<new_text>",'
+    '"occurrence_index": "<occurrence_index>"',
+)
+def update_file(
```
Comment (erik-megarad) [inline autogpt/commands/file_operations.py]: Hmm I'm not sure that the name of this command is reflective of what it does. I think `substitute_text` or something would be easier to understand.

File: autogpt/commands/file_operations.py [hunk 2]
```diff
@@ -224,6 +224,68 @@ def write_to_file(filename: str, text: str, config: Config) -> str:
...
+    filename: str, old_text: str, new_text: str, config: Config, occurrence_index=None
+):
+    """Update a file by replacing one or all occurrences of old_text with new_text using Python's built-in string
+    manipulation and regular expression modules for cross-platform file editing similar to sed and awk.
+
+    Args:
+        filename (str): The name of the file
+        old_text (str): String to be replaced. \n will be stripped from string.
```
Comment (erik-megarad) [inline autogpt/commands/file_operations.py]: Does the AI understand that by `String` we mean `String or regex`? Additionally, including some quotes around the `\n` would make it easier to read, it took me a few seconds to realize that it wasn't a bug in the docstring or something.

```suggestion
        old_text (str): String or regular expression to be replaced. "\n" will be stripped from string.
```

File: autogpt/commands/file_operations.py (hunk 1 above)
Comment (bfalans) [reply autogpt/commands/file_operations.py]: Discussed already, going to stick with update_file as it is consistent with other file commands.

File: autogpt/commands/file_operations.py (hunk 2 above)
Comment (bfalans) [reply autogpt/commands/file_operations.py]: method explicitely escapes old_string ito make sure it doesn't get misidentified as a regex.

Comment (erik-megarad) [reply autogpt/commands/file_operations.py]: Sorry I suppose I commented on the wrong line, `new_text` can be a regex

File: autogpt/commands/file_operations.py [hunk 3]
```diff
@@ -224,6 +225,68 @@ def write_to_file(filename: str, text: str, config: Config) -> str:
...
+        new_text (str): New string. \n will be stripped from string.
```
Comment (Pwuts) [inline autogpt/commands/file_operations.py]: \n will be stripped from **end of** string

File: autogpt/commands/file_operations.py (hunk 1 above)
Comment (Pwuts) [reply autogpt/commands/file_operations.py]: I agree with the suggestion to make the name more descriptive. `update_file` does not distinguish its function from `append_to_file`, but they definitely do different things.

How about `replace_in_file`?

**Review by merwanehamadi — APPROVED** [review]

**Comment by Pwuts** [comment]
Still some failing tests, but all seem unrelated to this change, so I say merge

**Review by Pwuts — APPROVED** [review]

**Pull request merged**